instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/InstrumenterBuilder.java

```diff
@@ -69,6 +72,45 @@ public final class InstrumenterBuilder<REQUEST, RESPONSE> {
   boolean propagateOperationListenersToOnEnd = false;
   boolean enabled = true;
 
+  private static final Map<String, List<ContextCustomizerCustomizer>> CONTEXT_CUSTOMIZER_MAP =
+      new HashMap<>();
+  private static final Map<String, List<AttributesExtractorCustomizer>> ATTRIBUTES_EXTRACTOR_MAP =
+      new HashMap<>();
+  private static final Map<String, List<OperationMetricsCustomizer>> OPERATION_METRICS_MAP =
+      new HashMap<>();
+
+  static {
+    ServiceLoader.load(ContextCustomizerCustomizer.class)
```

Comment:
Call during agent initialization, passing in a ServiceLoader based on AgentClassLoader
```suggestion
  static void init(ServiceLoader loader) {
    loader.load(ContextCustomizerCustomizer.class)
```